tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -472,4 +472,134 @@ TEST_F(
   }
 }
 
+class AGOverlapTest : public MultiDeviceTest {
+ protected:
+  OverlapTestParams params;
+
+  int64_t num_devices_;
+  int64_t my_device_index_;
+  std::vector<int64_t> all_devices_;
+  at::Tensor ta_, tb_, tc_;
+  at::Tensor ta_sharded_, tc_sharded_;
+  // stores the backend
+  c10d::Backend* world_communicator_;
+
+  // Define I/O and intermediate Tensor shapes
+  // A has shape (S, sharded(D), M/(S*D), K)
+  // B(K,N)
+  // C has shape (S, M/S, N)
+  std::vector<int64_t> ta_sizes;
+  std::vector<int64_t> tb_sizes;
+  std::vector<int64_t> tc_sizes;
+  std::vector<int64_t> ta_sharded_sizes;
+  std::vector<int64_t> tc_sharded_sizes;
+  std::vector<int64_t> tc_j_sizes;
+
+  void SetUp() {
+    MultiDeviceTest::SetUp();
+
+    num_devices_ = communicator_->size();
+    my_device_index_ = communicator_->deviceId();
+    ASSERT_EQ(params.M % num_devices_, 0);
+    ASSERT_EQ(params.S % num_devices_, 0);
```

Comment:
Why is this required?

Reply:
indeed, doesn't seem required. However we should make sure that `params.M  % (num_devices_ * params.S) == 0`